### ISSUE TITLE
Support for the experimental syntax 'flow' isn't currently enabled

### ISSUE THREAD

**Issue by dara-rockwell** [m-radzikowski/aws-sdk-client-mock#90]
**Checklist**

- [x] I have read [Caveats](https://github.com/m-radzikowski/aws-sdk-client-mock#caveats) documentation and didn't find a solution for this problem there.

**Bug description**

Our project does not use babel, so we're surprised to get this error when adding `aws-sdk-client-mock`. 

```
Test suite failed to run

    Jest encountered an unexpected token

    Jest failed to parse a file. This happens e.g. when your code or its dependencies use non-standard JavaScript syntax, or when Jest is not configured to support such syntax.

    Out of the box Jest supports Babel, which will be used to transform your files into valid JS based on your Babel configuration.

    By default "node_modules" folder is ignored by transformers.

    Here's what you can do:
     • If you are trying to use ECMAScript Modules, see https://jestjs.io/docs/ecmascript-modules for how to enable it.
     • If you are trying to use TypeScript, see https://jestjs.io/docs/getting-started#using-typescript
     • To have some of your "node_modules" files transformed, you can specify a custom "transformIgnorePatterns" in your config.
     • If you need a custom transformation specify a "transform" option in your config.
     • If you simply want to mock your non-JS modules (e.g. binary assets) you can stub them out with the "moduleNameMapper" config option.

    You'll find more details and examples of these config options in the docs:
    https://jestjs.io/docs/configuration
    For information about custom transformations, see:
    https://jestjs.io/docs/code-transformation

    Details:

    SyntaxError: /Code/project/node_modules/aws-sdk-client-mock/src/sinon.ts: Support for the experimental syntax 'flow' isn't currently enabled (3:8):

      1 | import {SinonStub} from 'sinon';
      2 |
    > 3 | export interface MaybeSinonProxy {
        |        ^
      4 |     isSinonProxy?: boolean;
      5 | }
      6 |

    Add @babel/preset-flow (https://git.io/JfeDn) to the 'presets' section of your Babel config to enable transformation.
    If you want to leave it as-is, add @babel/plugin-syntax-flow (https://git.io/vb4yb) to the 'plugins' section to enable parsing.

      at instantiate (node_modules/@babel/parser/src/parse-error/credentials.js:61:22)
      at toParseError (node_modules/@babel/parser/src/parse-error.js:58:12)
      at Parser.raise (node_modules/@babel/parser/src/tokenizer/index.js:1763:19)
      at Parser.expectOnePlugin (node_modules/@babel/parser/src/tokenizer/index.js:1827:18)
      at Parser.isExportDefaultSpecifier (node_modules/@babel/parser/src/parser/statement.js:2145:16)
      at Parser.maybeParseExportDefaultSpecifier (node_modules/@babel/parser/src/parser/statement.js:2007:14)
      at Parser.parseExport (node_modules/@babel/parser/src/parser/statement.js:1952:29)
      at Parser.parseStatementContent (node_modules/@babel/parser/src/parser/statement.js:431:25)
      at Parser.parseStatement (node_modules/@babel/parser/src/parser/statement.js:329:17)
      at Parser.parseBlockOrModuleBlockBody (node_modules/@babel/parser/src/parser/statement.js:1078:25)
```

**Environment**

- Node version: `v16.14.0`
- Testing lib and version: `jest 27.5.1`
- Typescript version: `4.5.5`
- AWS SDK v3 Client mock version: `0.6.2`
- AWS JS SDK libs and versions:
  - `@aws-sdk/client-s3 ^3.46.0`

**Jest Config**

```javascript
module.exports = {
  testEnvironment: 'node',
  collectCoverage: true,
  verbose: false,
  clearMocks: true,
  resetModules: true,
  restoreMocks: true,
  transform: {
    '^.+\\.ts$': 'ts-jest',
  },
  testRegex: '(test/.*)\\.test\\.ts$',
  coveragePathIgnorePatterns: ['test\\/.*'],
  modulePathIgnorePatterns: ['<rootDir>/test/__mocks__'],
  moduleFileExtensions: ['ts', 'js', 'json', 'node'],
  globals: {
    'ts-jest': {
      tsconfig: '<rootDir>/tsconfig.json',
    },
  },
  moduleDirectories: ['node_modules', 'src'],
  moduleNameMapper: {
    'test/(.*)': '<rootDir>/test/$1',
  },
};
```

**Comment by he-dslieker** [comment]
FYI: This occurs just by adding `aws-sdk-client-mock` as a dependency for the project without utilizing it at all.

**Comment by m-radzikowski** [comment]
@dara-rockwell please provide a repo with a reproduction, I used config you provided and had test run successful.

**Comment by dara-rockwell** [comment]
This is solved by #110
